src/modules/previewpane/powerpreview/settings.cpp

```diff
@@ -98,6 +100,11 @@ namespace PowerPreviewSettings
                             {
                                 Trace::PowerPreviewSettingsUpdateFailed(this->GetToggleSettingName().c_str(), lastState, newState, enabled);
                             }
+
+                            if (reinterpret_cast<ThumbnailProviderSettings*>(this))
```

Comment:
This interprets byte pattern in `this` pointer as another pointer type and will always evaluate to `true`. For runtime dispatch logic based on RTTI we need to use `dynamic_cast` instead.

Reply:
Thank you @yuyoyuppe 
I have updated the PR!